### PR TITLE
Automatic configurator service

### PR DESCRIPTION
**Pull request by kvaps**
This PR will implement additional confiscator container for linstor-satllite, it will do two main things:
- Add node to the cluster
- Define storage pools on the node

This code is going to reimplement the functionality added by https://github.com/kvaps/kube-linstor/pull/26 and https://github.com/kvaps/kube-linstor/pull/31

Fully supported resources list:
```
- controller -> properties
- resource-groups
- resource-groups -> select-filter
- resource-groups -> properties
- resource-groups -> volume-groups
- resource-groups -> volume-groups -> properties
- nodes
- nodes -> interfaces
- nodes -> properties
- nodes -> storage-pools
- nodes -> storage-pools -> properties
```

### CODE REVIEW

**Comment by kvaps** [comment]
@jbanety @t3hmrman the PR is ready for review.
It would be really nice if someone could confirm that it is working in your environment as good as mine.

Thank you!

**Comment by jbanety** [comment]
Hi @kvaps, 
I'll try this on a new cluster this morning.

**Comment by t3hmrman** [comment]
Hi @kvaps it might take me a while to find time to test this (hopefully by end of next week), will update here if/when I get to it!

File: helm/kube-linstor/scripts/functions.sh [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+  if $curl "$LS_CONTROLLERS/v1/nodes/${NODE_NAME}" >/dev/null; then
+    echo "Node $NODE_NAME already exists in cluster, skip adding..."
+    return 0
+  fi
+  echo "Node $NODE_NAME does not exists in cluster"
+
+  echo "Adding node $NODE_NAME to the cluster"
+  node_json="$(cat <<EOT
+{
+  "name": "",
```
Comment (jbanety) [inline helm/kube-linstor/scripts/functions.sh]: $NODE_NAME is missing here.
```shell
  "name": "$NODE_NAME",
```

File: helm/kube-linstor/templates/satellite-daemonset.yaml [hunk 2]
```diff
@@ -169,6 +131,80 @@ spec:
           mountPath: /run/lvm
         - name: run-udev
           mountPath: /run/udev
+      {{- if .Values.configurator.enabled }}
+      - name: configurator
+        {{- with .Values.satellite.image }}
+        image: "{{ .repository }}{{ if .digest }}@{{ .digest }}{{ else }}:{{ .tag }}{{ end }}"
+        imagePullPolicy: {{ .pullPolicy }}
+        {{- end }}
+        command:
```
Comment (jbanety) [inline helm/kube-linstor/templates/satellite-daemonset.yaml:140]: We need `privileged` context to display LVM volume groups.
```
        securityContext:
          privileged: true
```

**Review by jbanety — CHANGES_REQUESTED** [review]
Nice job !
Some fixes here.

**Comment by kvaps** [comment]
@jbanety thanks for review!

**Pull request merged**